File: DataStructuresAndLibraries/__3000__/3357/3357.py
class CircularList:
    def __init__(self, lst):
        self.lst = lst
        self.index = 0

    def next(self):
        if not self.lst:
            raise IndexError("The list is empty")
        item = self.lst[self.index]
        self.index = (self.index + 1) % len(self.lst)
        return item


# quantidade pessoas, litros, cabeça 
p, litros, cabaca = input().split(" ")
pessoas = CircularList(input().split(" "))

litros = float(litros)
cabaca = float(cabaca)
rico = ""

while litros >= cabaca:
  litros -= cabaca
  rico = pessoas.next()

rico = pessoas.next() #o último 

print(f"{rico} {litros:.1f}")

File: DataStructuresAndLibraries/__3000__/3357/3357.cpp
#include <iostream>
#include <vector>
#include <iomanip>
#include <string>

using namespace std;

class CircularLIst {

    private:
        vector<string> nomes;
        int index = 0;

    public:
        CircularLIst(vector<string> listadenomes){
            this,nomes = listadenomes;
        }

    string next() {
        if (this->nomes.empty()){
            thorw out_of_range('Lista de nomes vazia')
        }
        string nome = this->nomes[index];
        this->index = (this->index + 1) % this->nomes.size();
        return nome;
    }
};

int main(){

    vector<string> pessoas;
    int quantidade_de_pessoas;
    float litros, refill;
    string rico = '';

    cin >> quantidade_de_pessoas >> litros >> refill;

    for(int i = 0; i < quantidade_de_pessoas; i++){
        string nome;
        sin >> nome;
        pessoas.push_back(nome);
    }

    CircularLIst lista (pessoas);

    while(litros > refill){
        litros -= refill;
        rico = lista.next();
    }

    rico = lista.next(); // pega o nome do ultimo

    cout << rico << " " << fixed << setprecision(1) << litros << endl;
    
    return 0;
}